src/script/nix-prefetch-git

```diff
@@ -138,16 +138,7 @@ init_submodules(){
     # Add urls into .git/config file
     git submodule init
 
-    # list submodule directories and their hashes
-    git submodule status |
-    while read l; do
-        # checkout each submodule
-        local hash=$(echo $l | sed 's,^-\([0-9a-f]*\) \(.*\)$,\1,');
-        local dir=$(echo $l | sed 's,^-\([0-9a-f]*\) \(.*\)$,\2,');
-        local url=$(sed -n "\,$dir, { :loop; n; s,^.*url = ,,; T loop; p; q }" .git/config);
-
-        clone "$dir" "$url" "$hash" "";
-    done;
+    git submodule update --recursive
```

Comment:
It could, although I think the two specific steps provide more readability into the sequence of operations being performed.  Your suggestion is definitely valid `git` but I always feel like the `init` is a command and it's being hidden behind an option flag when doing it that way.  That's purely a stylistic perspective though: if I'm missing a technical reason why there's a difference I could easily be convinced to change my thinking here.